examples/two-columns-layout/src/utils.ts

```diff
@@ -0,0 +1,35 @@
+import { BaseItem, OnActiveParams } from '@algolia/autocomplete-core';
+import { dequal } from 'dequal/lite';
+
+type smartPreviewParams<TItem extends BaseItem> = OnActiveParams<TItem> & {
+  contextData: {
+    [key: string]: unknown;
+  };
+};
```

Comment:
Can we better type the `preview` property, since we know what data to expect?

```suggestion
type smartPreviewParams<TItem extends BaseItem> = AutocompleteScopeApi<TItem> & {
  state: AutocompleteState<TItem>
  preview: {
    [key: string]: unknown;
  };
};
```